src/main/java/org/jitsi/jigasi/transcription/AbstractTranscriptPublisher.java

```diff
@@ -450,26 +630,157 @@ BaseFormatter endedOn(TranscriptEvent event)
     public abstract class BasePromise
         implements Promise
     {
+        /**
+         * Whether {@link this#publish(Transcript)} has already been called once
+         */
+        private boolean published = false;
+
+        /**
+         * A unique directory name to store/publish the transcript into
+         */
+        private String dirName = generateHardToGuessTimeString("", "");
+
+        /**
+         * The fileName which will be used to record to the audio file to.
+         * Stays null when {@link this#wantsAudioRecording()} returns False.
+         */
+        private String audioRecordingFileName;
+
+        /**
+         * The recorder which will be used to record the audio, if required.
+         * Stays null when {@link this#wantsAudioRecording()} returns False.
+         */
+        private RecorderImpl recorder;
 
         @Override
         public boolean hasDescription()
         {
             return advertiseURL();
         }
 
+        /**
+         * Get the directory path wherein files can be stored, such as
+         * a representation of the {@link Transcript} which will need to be
+         * published
+         *
+         * @return the file name as a string
+         */
+        protected String getDirPath()
+        {
+            return dirName;
+        }
+
+        @Override
+        public boolean wantsAudioRecording()
+        {
+            return recordAudio();
+        }
+
+        @Override
+        public void giveAudioMixerMediaDevice(AudioMixerMediaDevice device)
```

Comment:
Can we:
1. Rename to something like `startRecording` (I originally suggested `setMediaDevice`, but the method actually starts the recording).
2. Use MediaDevice instead of AudioMixerMediaDevice
3. Use `MediaService.createRecorder(MediaDevice)` instead of creating the RecorderImpl instance directly